File: coffee-order-api/models/product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  }, 
  price: {
    type: Number,
    required: true,
  },
  imgURL: {
    type: String,
    required: true,
  },
  category: {
    type: mongoose.Schema.Types.ObjectId,
    ref:"Category",
    autopopulate: true,
  }
});

productSchema.plugin(require('mongoose-autopopulate'));

module.exports = mongoose.model('Product', productSchema);

File: coffee-order-terminal/app/src/main/java/com/feup/coffee_order_terminal/domain/model/Product.kt
package com.feup.coffee_order_terminal.domain.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Product(
    val _id: String,
    val name: String,
    val price: Double,
    val imgURL: String,
    val category: String
): Parcelable
